app/src/main/java/org/mozilla/fenix/wallpapers/Wallpaper.kt

```diff
@@ -4,13 +4,14 @@
 
 package org.mozilla.fenix.wallpapers
 
+import android.graphics.drawable.Drawable
 import org.mozilla.fenix.R
 
 /**
- * A enum that represents the available wallpapers and their states.
+ * A class that represents a available wallpaper and its state.
  */
-enum class Wallpaper(val drawable: Int, val isDark: Boolean) {
-    FIRST(drawable = R.drawable.wallpaper_1, isDark = true),
-    SECOND(drawable = R.drawable.wallpaper_2, isDark = false),
-    NONE(drawable = R.attr.homeBackground, isDark = false);
+data class Wallpaper(val name: String, val resource: Drawable? = null, val isDark: Boolean = false) {
+    companion object {
+        const val DEFAULT_RESOURCE = R.attr.homeBackground
+    }
```

Comment:
I don't think `resource` should be nullable here. It feels like the most essential part of the data, since we can't draw a `Wallpaper` without its resource.

Also, it might make sense to move `DEFAULT_RESOURCE` into the companion of `WallpaperManager` to keep all the static defintions grouped together.